core/src/program.rs

```diff
@@ -672,7 +701,7 @@ impl<EC: EvalCache> Program<EC> {
                         result.body.pos,
                     ))
                 }
-                _ => Ok(result.body),
+                _ => Ok(result.body.closurize(&mut vm.cache, result.env)),
```

Comment:
I wonder if  that can break stuff. In particular it seems `extract_from_term` is expecting the record spine to be fields with `Record` and `RecRecord` directly inside, while here it seems we're introducing intermediate `Closure` nodes everywhere.

PS: ok, I think I  get it. We're introducing closures only when the result isn't a record, which is treated above. It's still a strange discrepancy: it means that if you take a random value somewhere in the field of a record spine, if it's a record and you try to evaluate it further you might get unbound identifiers, but if it's not a record then it's properly closurized and it's evaluating fine.

I have mixed feelings about this, although it doesn't seem to break existing code. Intuitively I would expect `evaluate_record_spine` to ensure that the spine is just nested records with no `Closure`s in-between (something stupid: imagine we want to list all the accessible functions in the future, `eval_record_spine` would be a good candidate, but now functions are hidden behind `Closure` nodes, for example).

 I wonder if we should somehow specialize  `eval_record_spine` for doc testing to make it clear that it's a strange version that might not make sense for something else, and keep `eval_record_spine` with the current semantics (of course, under the hood, we can use a generic common base that just take an additional boolean-like argument to know if it should closurizes non record  or something like that). It's just a suggestion, there might be other ways to go about it,, but I have the overall feeling that we try to cram too many different things into one  method.